chore: config migration commands

hikchoi opened the pull request:
# chore: config migration commands
This PR 
- Introduces `IntermediateDendronConfig`, which has properties from both legacy and new configs.
- Adds migration that takes all command related legacy configs and consolidates into the new `commands` namespace
- Change implementation of all affected commands to respect both legacy and new configurations
  - When a new config is present, legacy config is ignored
- Adds metrics to track outstanding legacy configurations.

# Pull Request Checklist

You can go to [dendron pull requests](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html) to see full details for items in this checklist.

## General

### Quality Assurance
- [x] add a [test](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#writing-tests) for the new feature
- [x] make sure all the existing [tests](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#running-all-tests) pass
- [x] do a spot check by running your feature with our [test workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace)
- [ ] after you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: if you running mac/linux, check the windows output and vice versa if you are developing on windows

#### Special Cases
- [x] if your tests changes an existing snaphot, make sure that snapshots are [updated](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#updating-test-snapshots)
- [~] if you are adding a new language feature (graphically visible in vscode/preview/publishing), make sure that it is included in [test-workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace). We use this to manually inspect new changes and for auto regression testiing 

### Docs
- [x] Make sure that the PR title follows our [commit style](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [x] Please summarize the feature or impact in 1-2 lines in the PR description
- [ ] If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

hikchoi commented:
pinging @jonathanyeung for `randomNote` changes
pinging @Harshita-mindfire for `leaveTrace` addition to the new config namespace

hikchoi commented:
Will be adding docs for analytics changes and new configs shortly.

hikchoi commented:
NOTE:

The migration version should probably change since it's most likely not going to make it before v62 gets shipped.

Harshita-mindfire reviewed (approved):
leaveTrace looks good

hikchoi commented:
docs for analytics and config changes have been updated here: https://github.com/dendronhq/dendron-site/pull/221

hikchoi commented:
list of exhaustive diff in case we need it for the release note:

```
- lookup: LookupConfig;
  - New location: `DendronConfig.commands.lookup`
  - `lookup.note.selectionType` renamed to `lookup.note.selectionMode`
  - enum values of selection modes are renamed
    - `selectionExtract` renamed to `extract`
    - `selection2link` renamed to `link`
- lookupConfirmVaultOnCreate?: boolean;
  - Now part of `lookupConfig`
  - Renamed to `confirmVaultOnCreate`
  - Now a required property
  - Now explicitly defaults to `false`
  - New location: `DendronConfig.commands.lookup.note.confirmVaultOnCreate`
- insertNoteLink?: InsertNoteLinkConfig;
  - Now a required property
  - `aliasMode` now explicitly defaults to `none`
  - `enableMultiSelect` now explicitly defaults to `false`
  - New location: `DendronConfig.commands.insertNoteLink`
  - `insertNoteLinkConfig.multiselect` renamed to `insertNoteLink.enableMultiSelect`
- defaultInsertHierarchy?: string;
  - Now part of `insertNoteConfig`
  - renamed to `initialValue`
  - Explicitly defaults to `templates` but remains optional
  - New location: `DendronConfig.commands.insertNote.initialValue`
- randomNote?: RandomNoteConfig;
  - Now a required property
  - Properties `randomNote.include` and `randomNote.exclude` remains optional
  - New location: `DendronConfig.commands.randomNote`
- insertNoteIndex?: InsertNoteIndexConfig;
  - Now a required property
  - New location: `DendronConfig.commands.insertNoteIndex`
  - `insertNoteIndexConfig.marker` renamed to `insertNoteIndexConfig.enableMarker`
  - `insertNoteIndexConfig.enableMarker` is now a required property
  - `insertNoteIndexConfig.enableMarker` now explicitly defaults to `false`
```

hikchoi commented:
hmm looks like npm is down right now 😰
